### DISCUSSION TITLE
Mdx code block metadata not injected in props

### DISCUSSION THREAD

**Discussion by ElBartoTn** [gatsbyjs/gatsby#38106] [Help] [answered]
I'm trying to show code block from mdx file using react-live , when adding a metadata it's not being injected into the component as prop

  ```
 jsx live=true
  <button>Default</button>
```

when trying to log the props i only have this as result 
![image](https://github.com/gatsbyjs/gatsby/assets/33059027/0ae4e0bd-b0da-48ca-b024-a5a111826e25)

how can i properly access the `live` props 
I created this simple [example](https://github.com/ElBartoTn/gatsby-mdx)

**Comment by LekoArts** [comment, marked as answer]
This is expected behavior as per: https://mdxjs.com/guides/syntax-highlighting/#syntax-highlighting-with-the-meta-field

> @mdx-js/mdx doesn’t know whether you’re handling code as a component or what the format of that meta string is, so it defaults to how markdown handles it: meta is ignored.

You could use:
- https://github.com/remcohaszing/remark-mdx-code-meta
- https://github.com/LekoArts/gatsby-themes/tree/main/packages/rehype-meta-as-attributes
- Or write the utility on your own as described here: https://www.lekoarts.de/garden/adding-line-numbers-and-code-highlighting-to-mdx
